### PR TITLE
1151 hiv care & treatment enrolment form  

### LINKED ISSUE #1151: HIV Care & Treatment Enrolment Form - Multiple
**Path:** Dispensary > Patients > Add Program: `HIV Care & Treatment`

- [x] `Enrolment date`: unable to edit the enrolment date (default: today's date). User should be able to select a date in the past. 

Screen Shot 2023-02-08 at 5 05 19 PM

- [x] `Enrolment Patient Id` to be renamed `Program ID`

### **Mother:** 
- [x] `Clinic`: field to be removed from the form
#### `Mother's Patient ID`: 
- [ ] `HIV Program ID of the patient's mother`. Dependent field: it should only be visible and entered if patient is an infant (less than 2 years old). Since there is only one field in that section, I'd move it to the top section (see below): 

Screen Shot 2023-02-08 at 5 13 58 PM

`HIV Program ID of the patient's mother` - 2 options: 
- user manually enters the mother's program ID 
- user can select a Program ID in a Drop Down List that contains all Program ID of patients registered in this clinic (as a way to connect 2 patients records together). The infant's mother is always a registered patient with a HIV program ID. 

**Referral Details:** 

- [x] `Referral Details` section to be renamed to `Referral or Transfer in Details`
- [x] keep `Referred From:` as it is
- [x] keep `Priori ART:` as it is
- [x] Add `Clinic Transferred From:` (free text field)
- [x] Add `Previous Clinic ART Start Date:` (date picker, can be a date in the past)
- [x] Add `Date Transferred In` (date picker, can be a date in the past)
- [x] Add `Previous Clinic ID` (free text field)

For reference: 
Screen Shot 2023-02-08 at 5 24 22 PM

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #1151 

# 👩🏻‍💻 What does this PR do? 
- Enable enrolment date, so that the user can select a date in the past
- Enrolment Patient Id has been renamed to `Program ID`
- Clinic has been removed from Mother and now only has a `Mother's Program ID` field. 
- Added new sections to `Referral Details`

# 🧪 How has/should this change been tested? 
Try to enrol a patient into HIV Care and Treatment Program to see new fields + renames

### CODE REVIEW

File: server/service/src/sync/program_schemas/hiv_care_program.json [hunk 1]
```diff
@@ -96,28 +96,38 @@
...
-        },
         "hivConfirmationDate": {
           "format": "date",
           "type": "string"
         },
         "hivTestType": {
           "$ref": "#/definitions/HIVTestType"
         },
-        "mother": {
-          "$ref": "#/definitions/HIVRelation"
```
Comment (clemens-msupply) [inline server/service/src/sync/program_schemas/hiv_care_program.json:116]: I think we should keep the more general case of having the whole person object. In the UI schema you just record the mother id though. In this way we stay much more flexible. For example, if we need more person information in the future we already have a place to store it and we just need to change the UI.

File: server/service/src/sync/program_schemas/program_enrolment.json [hunk 2]
```diff
@@ -8,8 +8,8 @@
           "format": "date-time",
           "type": "string"
         },
-        "enrolmentPatientId": {
```
Comment (clemens-msupply) [inline server/service/src/sync/program_schemas/program_enrolment.json:11]: Thats a tricky one and I haven't found a good name yet, what it stores is "the patient id within a certain program". For example:

- HIVCare0001", "HIVCare0002
- HIVTesting0001, HIVTesting0002...

program ID sounds very much like the program name, e.g. HIVCareProgram and don't think its the right name.

Maybe `programEnrolmentId` or just `enrolmentId`

In the UI we can name it as it is best for the user though, i.e. as in the issue.

**Review by clemens-msupply — CHANGES_REQUESTED** [review]
In general, the json schema and the UI schema are independent and for the schema we should keep the more general case in mind when defining them, e.g. how to make it easy to capture additional data in the future. The UI schema can then be tweaked to only capture the required data. This is especially important for the base schemas which are used across all enrolments/encounters.

Otherwise it looks good!

File: server/service/src/sync/program_schemas/hiv_care_program.json (hunk 1 above)
Comment (roxy-dao) [reply server/service/src/sync/program_schemas/hiv_care_program.json]: both changes here (: thanks Clemens!
[1cf985e](https://github.com/openmsupply/open-msupply/pull/1215/commits/1cf985ed6388b04469981f45732477a20cfab228)

File: server/service/src/programs/program_enrolment/program_enrolment_updated.rs [hunk 3]
```diff
@@ -37,7 +37,7 @@ pub(crate) fn program_enrolment_updated(
         name: document.name.clone(),
         patient_id: patient_id.to_string(),
         enrolment_datetime,
-        program_patient_id: program.enrolment_patient_id,
+        program_patient_id: program.program_enrolment_id,
```
Comment (clemens-msupply) [inline server/service/src/programs/program_enrolment/program_enrolment_updated.rs:40]: Could you please rename the DB row as well? (just edit the initial migration)

Comment (clemens-msupply) [reply server/service/src/programs/program_enrolment/program_enrolment_updated.rs:40]: And probably filters and sort (if there are any)...

Comment (roxy-dao) [reply server/service/src/programs/program_enrolment/program_enrolment_updated.rs]: [59b9e0f](https://github.com/openmsupply/open-msupply/pull/1215/commits/59b9e0feb385f7e93bd9506fbf5093b1264f75d8)

**Review by clemens-msupply — APPROVED** [review]
Thanks you! looks good!

**Pull request merged**